syz-cluster/email-reporter/sender.go

```diff
@@ -3,20 +3,112 @@
 
 package main
 
-import "context"
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net/smtp"
+	"strconv"
+	"strings"
+
+	"github.com/google/syzkaller/syz-cluster/pkg/app"
+	"github.com/google/uuid"
+)
 
 // TODO: how can we test it?
 // Using some STMP server library is probably an overkill?
 
 type smtpSender struct {
+	cfg      *app.EmailConfig
+	host     string
+	port     int
+	user     string
+	password string
+}
+
+func newSender(ctx context.Context, cfg *app.EmailConfig, secretManager app.SecretManager) (*smtpSender, error) {
+	values := map[app.SecretKey]string{}
+	for _, key := range []app.SecretKey{
+		app.SecretSMTPHost, app.SecretSMTPPort, app.SecretSMTPUser, app.SecretSMTPPassword,
+	} {
+		val, err := secretManager.Get(ctx, key)
+		if err != nil {
+			return nil, fmt.Errorf("failed to query %v: %w", val, err)
+		}
+	}
+	port, err := strconv.Atoi(values[app.SecretSMTPPort])
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse SMTP port: not a valid integer")
+	}
+	return &smtpSender{
+		cfg:      cfg,
+		host:     values[app.SecretSMTPHost],
+		port:     port,
+		user:     values[app.SecretSMTPUser],
+		password: values[app.SecretSMTPPassword],
+	}, nil
 }
 
 // Send constructs a raw email from EmailToSend and sends it over SMTP.
 func (sender *smtpSender) Send(ctx context.Context, item *EmailToSend) (string, error) {
-	// TODO:
-	// 1) Fill in email headers, including the Message ID.
-	// https://pkg.go.dev/github.com/emersion/go-message/mail#Header.GenerateMessageIDWithHostname
-	// 2) Send over STMP:
-	// https://pkg.go.dev/net/smtp
-	return "", nil
+	uuidVal, err := uuid.NewRandom()
+	if err != nil {
+		return "", fmt.Errorf("failed to generate UUID: %w", err)
+	}
+
+	msgID := fmt.Sprintf("<%s@%s>", uuidVal, sender.host)
```

Comment:
You can https://pkg.go.dev/github.com/google/uuid#NewString here w/o error.
It saves 5 lines.